scvi/module/base/_base_module.py

```diff
@@ -523,18 +533,31 @@ def _generic_forward(
     get_inference_input_kwargs,
     get_generative_input_kwargs,
     compute_loss,
+    latent_data_type,
 ):
     """Core of the forward call shared by PyTorch- and Jax-based modules."""
+    is_latent = latent_data_type is not None
+    if is_latent and compute_loss:
+        raise ValueError("compute_loss currently not supported in latent mode forward")
+
     inference_kwargs = _get_dict_if_none(inference_kwargs)
     generative_kwargs = _get_dict_if_none(generative_kwargs)
     loss_kwargs = _get_dict_if_none(loss_kwargs)
     get_inference_input_kwargs = _get_dict_if_none(get_inference_input_kwargs)
     get_generative_input_kwargs = _get_dict_if_none(get_generative_input_kwargs)
 
+    if is_latent:
+        get_inference_input_kwargs["latent_data_type"] = latent_data_type
+
     inference_inputs = module._get_inference_input(
         tensors, **get_inference_input_kwargs
     )
-    inference_outputs = module.inference(**inference_inputs, **inference_kwargs)
+    if is_latent:
+        inference_outputs = module.inference_no_encode(
+            **inference_inputs, **inference_kwargs
+        )
+    else:
```

Comment:
I think it would be best not to include this new function, so as to isolate these changes to just changes within the respective model/module classes.

Inference in the VAE could look like this:

```python
def inference(...):
    if dist_params is not None:
        self._inference_no_encode(...)
    else:
        self._inference_with_encode(...)
```

The way the base module is written already permits this flexibility. Once the anndatamanager is registered with the latent fields it will start adding these tensors into the minibatches, so this is a way for the module to know which one to route to.